internal/app/app.go

```diff
@@ -0,0 +1,123 @@
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+)
+
+// set of exported json options
+const (
+	ExportedJSONPrefix = ""
+	ExportedJSONIndent = "    "
+)
+
+// Data is the exported Realm app config data which it is identified by
+type Data struct {
+	ID   string `json:"client_app_id,omitempty"`
+	Name string `json:"name"`
+}
+
+func (d Data) IsZero() bool {
+	return d.Name == ""
+}
```

Comment:
i felt it was worth it since `IsZero` contains specific logic other than "are all my fields zero?" ... and its usage will be more than just one place, we just haven't built those commands out yet.  for instance, in `push` code that I have locally right now:

```go
appData, appDataErr := app.ResolveData(profile.WorkingDirectory)
if appDataErr != nil {
	return appDataErr
}

if i.AppDirectory == "" {
	if appData.IsZero() {
		return errProjectNotFound{flagAppDirectory}
	}
	i.AppDirectory = profile.WorkingDirectory
}
```

happy to remove this and simply go with `appData.Name == ""` everywhere, but personally I like to see that kind of logic owned by the `struct` so future devs won't have to know we need to look at `Name` (and not `ID`) to verify whether or not this is valid app data

Reply:
Most devs will see that we are checking `appData.Name` so I don't see this as an issue. My concern with having this `IsZero` is that future devs won't take the extra step to see what it means for an app to be zero (aka the name is missing). This abstraction seems more harm than good since it's a one liner either way.

Reply:
all fair points, and again no strong opinions for this on my end.  i'll update accordingly